app/src/main/java/com/zgamelogic/ghosty/InvestigationViewManager.java

```diff
@@ -0,0 +1,272 @@
+     */
+    public void populateEvidenceList() {
+        RelativeLayout itemView;
+        LayoutInflater inflater;
+        TextView textview;
+        Switch switchview;
+
+        // Clear evidenceList of any existing items
+        evidenceList.removeAllViews();
+
+        // Add each evidence back in as a default state
+        inflater = activity.getLayoutInflater();
+        for (String ev : evidences) {
+            itemView = (RelativeLayout)inflater.inflate(R.layout.view_item, evidenceList, false);
+
+            // Textview
+            textview = (TextView)itemView.findViewById(R.id.evidenceListString);
+            textview.setText(ev);
+            textview.setVisibility(View.VISIBLE);
+
+            // Switch
+            switchview = (Switch)itemView.findViewById(R.id.evSwitch);
+            switchview.setOnCheckedChangeListener((compoundButton, isChecked) -> {
+                RelativeLayout evidenceLayout;
+                TextView evidenceTextview;
+                String evidence;
+                HashSet<String> remainingEv;
+                Log.i(LOG_IVM, "Clicked");
+
+                evidenceLayout = (RelativeLayout) compoundButton.getParent();
+                evidenceTextview = (TextView) evidenceLayout.findViewById(R.id.evidenceListString);
+                evidence = (String) evidenceTextview.getText();
+
+                // Update list of selected evidences
+                if (isChecked) {
+                    Log.i(LOG_IVM, "isChecked");
+                    selectedEvidences.add(evidence);
+                    Log.i(LOG_IVM, String.valueOf(selectedEvidences.contains(evidence)));
+
+                } else {
+                    Log.i(LOG_IVM, "notChecked");
+                    selectedEvidences.remove(evidence);
+                }
+
+                // Update ghosts on the UI
+                remainingEv = updateGhostLayout();
+                Log.i(LOG_IVM, "updated");
+
+                // Update evidences on the UI
+                updateEvidenceLayout(remainingEv);
+
+
+            });
+            switchview.setVisibility(View.VISIBLE);
+
+            // Add evidence to layout
+            evidenceList.addView(itemView);
+        }
+    }
+
+
+    /**
+     * Populate the ghost list based on ghosts.
+     *
+     * This is intended to be called once by MainActivity's getGhosts method
+     * as this will remake the list from scratch.
+     */
+    public void populateGhostList() {
+        RelativeLayout itemView;
+        LinearLayout ghostEvidenceList;
+        LayoutInflater inflater;
+        LinearLayout.LayoutParams layoutParams;
+        TextView textview;
+        HashSet<String> evidences;
+
+        // Clear ghostList of any existing items
+        ghostList.removeAllViews();
+
+        /* Need to manually set the layout parameters for each evidence
+           that is added to the ghost's ui list of evidences. */
+        layoutParams = new LinearLayout.LayoutParams(0,
```

Comment:
Get this on one line or make it 5
```suggestion
        layoutParams = new LinearLayout.LayoutParams(
            0,
            LinearLayout.LayoutParams.WRAP_CONTENT,
            1
         );
```